examples/by_feature/checkpointing.py

```diff
@@ -155,7 +155,7 @@ def training_function(config, args):
     model = model.to(accelerator.device)
 
     # Instantiate optimizer
-    optimizer = AdamW(params=model.parameters(), lr=lr, correct_bias=correct_bias)
+    optimizer = AdamW(params=model.parameters(), lr=lr, correct_bias=correct_bias, no_deprecation_warning=True)
```

Comment:
Switch to the torch `AdamW`? The HF ones as small bugs that can hurt larger trainings.